Jenkinsfile

```diff
@@ -0,0 +1,166 @@
+def registry = '562955126301.dkr.ecr.eu-west-2.amazonaws.com'
+def regCredsId = 'ecr:eu-west-2:ecr-user'
+def kubeCredsId = 'awskubeconfig002'
+def ingressServer = "ffc.aws-int.defra.cloud"
+def imageName = 'ffc-demo-claim-service'
+def repoName = 'ffc-demo-claim-service'
+def repoUrl = ''
+def commitSha = ''
+def branch = ''
+def pr = ''
+def mergedPrNo = ''
+def containerTag = ''
+
+def getMergedPrNo() {
+    def mergedPrNo = sh(returnStdout: true, script: "git log --pretty=oneline --abbrev-commit -1 | sed -n 's/.*(#\\([0-9]\\+\\)).*/\\1/p'").trim()
+    return mergedPrNo ? "pr$mergedPrNo" : ''
+}
+
+def getRepoURL() {
+  return sh(returnStdout: true, script: "git config --get remote.origin.url").trim()
+}
+
+def getCommitSha() {
+  return sh(returnStdout: true, script: "git rev-parse HEAD").trim()
+}
+
+def getVariables(repoName) {
+    def branch = BRANCH_NAME
+    // use the git API to get the open PR for a branch
+    // Note: This will cause issues if one branch has two open PRs
+    def pr = sh(returnStdout: true, script: "curl https://api.github.com/repos/DEFRA/$repoName/pulls?state=open | jq '.[] | select(.head.ref == \"$branch\") | .number'").trim()
+    def rawTag = pr == '' ? branch : "pr$pr"
+    def containerTag = rawTag.replaceAll(/[^a-zA-Z0-9]/, '-').toLowerCase()
+    return [branch, pr, containerTag,  getMergedPrNo(), getRepoURL(), getCommitSha()]
+}
+
+def updateGithubCommitStatus(message, state, repoUrl, commitSha) {
+  step([
+    $class: 'GitHubCommitStatusSetter',
+    reposSource: [$class: "ManuallyEnteredRepositorySource", url: repoUrl],
+    commitShaSource: [$class: "ManuallyEnteredShaSource", sha: commitSha],
+    errorHandlers: [[$class: 'ShallowAnyErrorHandler']],
+    statusResultSource: [ $class: "ConditionalStatusResultSource", results: [[$class: "AnyBuildResult", message: message, state: state]] ]
+  ])
+}
+
+def buildTestImage(name, suffix) {
+  sh 'docker image prune -f || echo could not prune images'
+  // NOTE: the docker-compose file currently makes use of global $BUILD_NUMBER env vars fo image names
+  sh "docker-compose -p $name-$suffix -f docker-compose.yaml -f docker-compose.test.yaml build --no-cache $name"
+}
+
+def runTests(name, suffix) {
+  try {
+    sh 'mkdir -p test-output'
+    sh 'chmod 777 test-output'
+    sh "docker-compose -p $name-$suffix -f docker-compose.yaml -f docker-compose.test.yaml up $name"
```

Comment:
Not important, but the `run` command feels more natural than `up` when we're running a task, as opposed to starting up a persistent service.